modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/utils/bean/APICreationRequestBean.java

```diff
@@ -0,0 +1,342 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.am.integration.test.utils.bean;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.wso2.am.integration.test.utils.APIManagerIntegrationTestException;
+
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Bean class to contain the Information needed to create and update process of a API.
+ */
+public class APICreationRequestBean extends AbstractRequest {
+
+    private static final Log log = LogFactory.getLog(APIRequest.class);
+    private String name;
+    private String context;
+    private JSONObject endpoint;
+    private String version;
+    private String visibility = "public";
+    private String description = "description";
+    private String endpointType = "nonsecured";
+    private String http_checked = "http";
+    private String https_checked = "https";
+    private String tags = "tags";
+    private String tier = "Silver";
+    private String thumbUrl = "";
+    private String tiersCollection = "Gold";
+    private String resourceCount = "0";
+    private String roles = "";
+    private String wsdl = "";
+    private String default_version = "";
+    private String default_version_checked = "";
+    private List<APIResourceBean> resourceBeanList;
+    private String epUsername = "";
+    private String epPassword = "";
+    private String sandbox = "";
+    private String provider = "admin";
```

Comment:
AFAIU according to the API call [1] its the creator of API.

[1] action=updateStatus&name=xxx&version=1.0.0&*provider=apiCreateName*
&status=PUBLISHED&publishToGateway=true&requireResubscription=true

On Thu, Apr 2, 2015 at 11:50 AM, Krishantha Samaraweera <
notifications@github.com> wrote:

> In
> modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/utils/bean/APICreationRequestBean.java
> <https://github.com/wso2/product-apim/pull/67#discussion_r27634633>:
>
> > +    private String http_checked = "http";
> > +    private String https_checked = "https";
> > +    private String tags = "tags";
> > +    private String tier = "Silver";
> > +    private String thumbUrl = "";
> > +    private String tiersCollection = "Gold";
> > +    private String resourceCount = "0";
> > +    private String roles = "";
> > +    private String wsdl = "";
> > +    private String default_version = "";
> > +    private String default_version_checked = "";
> > +    private List<APIResourceBean> resourceBeanList;
> > +    private String epUsername = "";
> > +    private String epPassword = "";
> > +    private String sandbox = "";
> > +    private String provider = "admin";
>
> Any reason to set provider as "admin" always ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wso2/product-apim/pull/67/files#r27634633>.
>



-- 
Waruna Perera
Senior Software Engineer - Test Automation
Mobile: +94 77 3867037
WSO2, Inc.; http://wso2.com/
lean . enterprise . middlewear.